resource/wait/port/port.go

```diff
@@ -0,0 +1,90 @@
+// Copyright © 2016 Asteris, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package port
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/asteris-llc/converge/resource"
+	"github.com/asteris-llc/converge/resource/wait"
+	"github.com/pkg/errors"
+)
+
+const (
+	defaultInterval = 5 * time.Second
+	defaultTimeout  = 10 * time.Second
+	defaultRetries  = 5
+	defaultHost     = "localhost"
+)
+
+// Port represents a port check
+type Port struct {
+	*resource.Status
+	*wait.Retrier
+	Host string
+	Port int
+}
+
+// Check if the port is open
+func (p *Port) Check(resource.Renderer) (resource.TaskStatus, error) {
+	p.Status = resource.NewStatus()
+
+	alive, err := p.checkConnection()
+	if err != nil {
+		return p, errors.Wrapf(err, "failed to check connection")
```

Comment:
actually this means something other than a network error (net.OpErr) occurred. I don't have an example of what this may be at this point as I've never seen anything else yet and don't see any other error types being returned after a cursory look through the stdlib code. If the connection fails b/c of an OpErr, checkConnection returns false and nil for the error.